language/move-core/types/src/vm_status.rs

```diff
@@ -554,6 +554,8 @@ pub enum StatusCode {
     // The sender is trying to publish a module named `M`, but the sender's account already
     // contains a module with this name.
     DUPLICATE_MODULE_NAME = 1095,
+    // The sender is trying to publish a module that breaks the compatibility checks
+    INCOMPATIBLE_MODULE = 1096,
```

Comment:
Nit: `BACKWARD_INCOMPATIBLE_MODULE_UPGRADE` (or similar) might be a bit more explicit?